app/src/main/java/edu/unikom/herbamedjabar/migration/ScanHistoryMigrationManager.kt

```
@@ -0,0 +1,40 @@
+                for (history in allHistory) {
+                    val parsed = PlantDataParser.parsePlantData(history.resultText)
+                    val updatedHistory = history.copy(
+                        plantName = parsed["plantName"] ?: "",
+                        content = parsed["description"] ?: "",
+                        benefit = parsed["benefit"] ?: "",
+                        warning = parsed["warning"] ?: ""
+                    )
+                    scanHistoryDao.updateHistory(updatedHistory)
+                }
+                prefs.edit { putBoolean("scan_history_migrated_v2", true) }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)